### PR TITLE
New Resource: `azurerm_iothub_consumer_group`

### LINKED ISSUE #2230: Support for Iot Hub consumer groups
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Support is already provided for Event Hub consumer groups ([`azurerm_eventhub_consumer_group`](https://www.terraform.io/docs/providers/azurerm/r/eventhub_consumer_group.html)). Although IoT Hubs provide an Event Hub-compatible endpoint for message events, it's not clear that the `azurerm_eventhub_consumer_group` supports these. I was not able to either create a new consumer group on an IoT Hub, or import an existing one with this resource.

Event Hub consumer groups appear to have IDs [1] of a different form to IoT Hub consumer groups [2].

### New or Affected Resource(s)

* azurerm_eventhub_consumer_group
* azurerm_iothub
* (proposed new) azurerm_iothub_consumer_group

### Potential Terraform Configuration

```hcl
resource "azurerm_iothub_consumer_group" "test" {
  name                = "test"
  iothub_name         = "${azurerm_iothub.test.name}"
  resource_group_name = "${azurerm_resource_group.test.name}"
}
```

### References

[1] https://docs.microsoft.com/en-ca/rest/api/eventhub/consumergroups/get
[2] https://docs.microsoft.com/en-ca/rest/api/iothub/iothubresource/geteventhubconsumergroup

### PR DESCRIPTION
**Pull request by tombuildsstuff**
Tests pass:
```
$ acctests azurerm TestAccAzureRMIotHubConsumerGroup_
=== RUN   TestAccAzureRMIotHubConsumerGroup_events
--- PASS: TestAccAzureRMIotHubConsumerGroup_events (280.43s)
=== RUN   TestAccAzureRMIotHubConsumerGroup_operationsMonitoringEvents
--- PASS: TestAccAzureRMIotHubConsumerGroup_operationsMonitoringEvents (211.24s)
PASS
ok  	github.com/terraform-providers/terraform-provider-azurerm/azurerm	493.476s
```

Fixes #2230

### CODE REVIEW

File: azurerm/resource_arm_iothub_consumer_group_test.go [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+		return nil
+
+	}
+}
+
+func testAccAzureRMIotHubConsumerGroup_basic(rInt int, location, eventName string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "foo" {
+  name = "acctestRG-%d"
```
Comment (katbyte) [inline azurerm/resource_arm_iothub_consumer_group_test.go]: Misaligned assignments here

File: azurerm/resource_arm_iothub_consumer_group_test.go [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+  location = "%s"
+}
+
+resource "azurerm_iothub" "test" {
+  name                = "acctestIoTHub-%d"
+  resource_group_name = "${azurerm_resource_group.foo.name}"
+  location            = "${azurerm_resource_group.foo.location}"
+  sku {
+    name = "B1"
+    tier = "Basic"
```
Comment (katbyte) [inline azurerm/resource_arm_iothub_consumer_group_test.go]: And here

File: azurerm/resource_arm_iothub_consumer_group_test.go (hunk 1 above)
Comment (tombuildsstuff) [reply azurerm/resource_arm_iothub_consumer_group_test.go]: 👍🏻

File: azurerm/resource_arm_iothub_consumer_group_test.go (hunk 2 above)
Comment (tombuildsstuff) [reply azurerm/resource_arm_iothub_consumer_group_test.go]: 👍🏻

**Pull request merged**